haystack/components/converters/csv.py

```diff
@@ -102,3 +105,231 @@ def run(
             documents.append(document)
 
         return {"documents": documents}
+
+
+@component
+class CSVDocumentSplitter:
+    """
+    A component to split a CSV document into multiple CSV documents
+
+    It is based on:
+    1) Consecutive empty rows (the "classic" approach).
+    2) Optionally detecting side-by-side tables (via a BFS search for non-empty cells).
+
+    Attributes:
+        split_threshold (int): The number of consecutive empty rows required to split the CSV. Default is 2.
+        delimiter (str): The delimiter used in the CSV. Default is ','.
+        quotechar (str): The quote character used in the CSV. Default is '"'.
+        trim_empty_rows (bool): Whether to trim leading/trailing empty rows in each block. Default is True.
+        skip_errors (bool): Whether to skip documents with parsing errors. Default is True.
+        detect_side_tables (bool): Whether to detect tables that sit side-by-side in different columns.
+                                   If True, a more expensive BFS-based approach is used.
+    """
+
+    def __init__(
+        self,
+        split_threshold: int = 2,
+        delimiter: str = ",",
+        quotechar: str = '"',
+        trim_empty_rows: bool = True,
+        skip_errors: bool = True,
+        detect_side_tables: bool = False,
+    ) -> None:
+        if split_threshold < 1:
+            raise ValueError("split_threshold must be at least 1")
+        self.split_threshold = split_threshold
+        self.delimiter = delimiter
+        self.quotechar = quotechar
+        self.trim_empty_rows = trim_empty_rows
+        self.skip_errors = skip_errors
+        self.detect_side_tables = detect_side_tables
+
+    def _split_csv_content(self, content: str) -> List[str]:
+        """Splits CSV content into blocks of CSV text."""
+        try:
+            reader = csv.reader(
+                io.StringIO(content), delimiter=self.delimiter, quotechar=self.quotechar, skipinitialspace=True
+            )
```

Comment:
I've often run into situations where I get a mixture of CSV files with different types of delimiters and sometimes quotecharacters. So often times having a global setting like this isn't ideal. So I have two ideas/approaches: 
1. Use pandas.read_csv instead. It is much heavier but it is what we are often using in our other components that handle CSVs and Excel files. 
2. Make using the `csv` library more robust. For example, use the `csv.Sniffer` (also used by Pandas) to guess the dialect. In addition to this I'd set delimiter and quotechar to None by default. 
```
dialect = csv.Sniffer().sniff(csvfile.readline(), [',',';'])
csvfile.seek(0)  
data = csv.reader(csvfile, dialect)
```